assessmentModel/src/commonMain/kotlin/recorders/MotionRecorderConfiguration.kt

```diff
@@ -0,0 +1,142 @@
+}
+
+/**
+ * [MotionRecorderType] is used to enumerate the sensors and calculated measurements that should be recorded for a
+ * given assessment.
+ *
+ * Splitting the device motion into components in this manner stores the data using a consistent JSON schema that
+ * can represent the sensor data returned by both iOS and Android devices. Additionally, this allows for representing
+ * the data using a single table which can then be  filtered by type to perform calculations and DSP on the input
+ * sources.
+ *
+ * The recorder types are split into two categories of [raw] sensor data and [calculated] values. The [calculated]
+ * values are are black box values calculated by the platform libraries and may vary between devices.
+ *
+ * - Raw data:
+ *   - [Accelerometer]: Raw accelerometer reading.
+ *   - [Gyro]: Raw gyroscope reading.
+ *   - [Magnetometer]: Raw magnetometer reading.
+ *
+ * - Calculated values:
+ *   - [Attitude]: Calculated orientation of the device using the gyro and magnetometer (if appropriate).
+ *   - [Gravity]: Calculated vector for the direction of gravity in the coordinates of the device.
+ *   - [MagneticField]: The magnetic field vector with respect to the device for devices with a magnetometer.
+ *   - [RotationRate]: The rotation rate of the device for devices with a gyro.
+ *   - [UserAcceleration]: Calculated vector for the participant's acceleration in the coordinates of the device.
+ *
+ */
+@Serializable
+enum class MotionRecorderType : StringEnum {
+    Accelerometer, Gyro, Magnetometer,
+    Attitude, Gravity, MagneticField, RotationRate, UserAcceleration;
+
+    @Serializer(forClass = MotionRecorderType::class)
+    companion object : KSerializer<MotionRecorderType> {
+        override val descriptor: SerialDescriptor = PrimitiveDescriptor("MotionRecorderType", PrimitiveKind.STRING)
+        override fun deserialize(decoder: Decoder): MotionRecorderType {
+            val name = decoder.decodeString()
+            return values().matching(name) ?: throw SerializationException("Unknown $name for ${descriptor.serialName}. Needs to be one of ${values()}")
+        }
+        override fun serialize(encoder: Encoder, value: MotionRecorderType) {
+            encoder.encodeString(value.name.decapitalize())
+        }
+
+        val raw = setOf(
+            Accelerometer,
+            Gyro,
+            Magnetometer
+        )
+        val calculated = setOf(
+            Attitude,
+            Gravity,
+            MagneticField,
+            RotationRate,
+            UserAcceleration
+        )
+    }
+}
+
+/**
+ * A [MotionRecord] is a serializable implementation of a [SampleRecord] that can be used to record a sample from one
+ * of the motion sensors or calculated vectors. All properties on the record ard defined as `null` to allow using this
+ * same serialization object to define both path markers that are added to the data stream to mark step transitions
+ * and the results of the sensor.
```

Comment:
Suggested for the last sentence:

Path markers are added to the data stream to mark step transitions using this same serialization object by setting all properties on the record to `null`.